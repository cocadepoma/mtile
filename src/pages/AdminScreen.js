import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';

import { fetchWithToken } from '../helpers/fetch';
import validator from 'validator';
import Switch from "react-switch";

import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { ModalToastify } from '../components/ui/ModalToastify';
import { AdminModal } from '../components/admin/AdminModal';
import { uiOpenModal } from '../actions/ui';
import { disableScroll } from '../helpers/disable-enable-scroll';
import { startDeleteOrderEvent } from '../actions/calendar';
import { ToastSuccess } from '../components/ui/ToastSuccess';
import { ToastError } from '../components/ui/ToastError';
import { startAddUser, startDeleteUser } from '../actions/auth';

export const AdminScreen = () => {

    const [formValuesUser, setFormValuesUser] = useState({ email: '', name: '', password: '' });
    const { email, name, password } = formValuesUser;

    const [formValuesOrder, setFormValuesOrder] = useState({ orderId: '' });
    const { orderId } = formValuesOrder;

    const [users, setUsers] = useState([]);
    const [isAdmin, setIsAdmin] = useState(false);
    const [idUser, setIdUser] = useState(null);

    const dispatch = useDispatch();

    useEffect(() => {
        async function loadUsers() {
            const resp = await fetchWithToken('users/', undefined, 'GET');
            const { users } = await resp.json();
            if (users) {
                setUsers([...users]);
            }
        }
        loadUsers();
    }, [])


    const handleInputChangeUser = ({ target }) => {
        const name = target.name;
        document.querySelector(`input[name="${name}"]`).classList.remove('border-red');

        setFormValuesUser({
            ...formValuesUser,
            [target.name]: target.value
        });
    }

    const handleInputSwitch = () => {
        setIsAdmin(!isAdmin)
    }

    // Will call tostify first to confirm the option the user will choose. Cancel or Deny.
    const handleStartDeleteUser = (id, name) => {
        toast.warn(<ModalToastify
            handleDeleteItem={() => handleDeleteUser(id)}
            code={name}
            message="Estás seguro de borrar el usuario" />,
            {
                position: toast.POSITION.TOP_CENTER,
                closeOnClick: false,
                autoClose: false,
                toastId: '1'
            });
    }

    const handleDeleteUser = async (id) => {
        const resp = await dispatch(startDeleteUser(id));

        if (resp?.ok) {
            setUsers(users.map(user => user.id === resp.user.id ? resp.user : user));
            toast.success(<ToastSuccess text={resp.msg} />, { position: toast.POSITION.TOP_CENTER });
        } else {
            toast.error(<ToastError text={resp.msg} />, { position: toast.POSITION.TOP_CENTER });
        }
    }

    const handleUpdateUser = (id) => {
        setIdUser(id);
        disableScroll();
        dispatch(uiOpenModal());
    }

    const handleSubmitUser = async (e) => {
        e.preventDefault();
        e.stopPropagation();

        let isValid = true;
        let message = '';

        if (!validator.isEmail(email)) {
            document.querySelector('input[name="email"]').classList.add('border-red');
            isValid = false;
            message += 'El email debe de ser un email válido. \n';
        } else {
            document.querySelector('input[name="email"]').classList.remove('border-red');
        }

        if (name.trim().length < 4) {
            document.querySelector('input[name="name"]').classList.add('border-red');
            isValid = false;
            message += 'El nombre debe de tener al menos 4 caracteres. \n';
        } else {
            document.querySelector('input[name="name"]').classList.remove('border-red');
        }

        if (password.trim().length < 6) {
            document.querySelector('input[name="password"]').classList.add('border-red');
            isValid = false;
            message += 'La password debe de tener al menos 4 caracteres. \n';
        } else {
            document.querySelector('input[name="password"]').classList.remove('border-red');
        }

        if (!isValid) {
            return toast.error(message, { position: toast.POSITION.TOP_CENTER });
        }

        const resp = await dispatch(startAddUser({ ...formValuesUser, admin: isAdmin }))

        if (resp?.ok) {
            setUsers([...users, resp.user]);
            toast.success(<ToastSuccess text={resp.msg} />, { position: toast.POSITION.TOP_CENTER });
            setFormValuesUser({ email: '', name: '', password: '' });
            setIsAdmin(false);
        } else {
            const message = resp.msg || 'Error al crear el usuario, inténtelo más tarde';
            toast.error(<ToastError text={message} />, { position: toast.POSITION.TOP_CENTER });
        }

    }

    const handleInputChangeOrder = ({ target }) => {

        document.querySelector(`input[name="${target.name}"]`).classList.remove('border-red');
        setFormValuesOrder({
            ...formValuesUser,
            [target.name]: target.value
        });
    }

    // Will call tostify first to confirm the option the user will choose. Cancel or Deny.
    const handleStartDeleteOrder = (e) => {
        e.preventDefault();
        e.stopPropagation();

        if (orderId !== '') {
            toast.warn(<ModalToastify
                handleDeleteItem={handleSubmitOrder}
                code={orderId}
                message="Estás seguro de borrar la order con el número" />,
                {
                    position: toast.POSITION.TOP_CENTER,
                    closeOnClick: false,
                    autoClose: false,
                    toastId: '1'
                });
        } else {
            document.querySelector('input[name="orderId"]').classList.add('border-red');
            return toast.error('Debes indicar el número de orden', { position: toast.POSITION.TOP_CENTER });
        }
    }

    const handleSubmitOrder = async () => {

        let isValid = true;
        let message = '';

        if (!Number.isInteger(Number(orderId)) || orderId.trim().length === 0 || !validator.isNumeric(orderId)) {
            document.querySelector('input[name="orderId"]').classList.add('border-red');
            isValid = false;
            message += 'El id es obligatorio y debe de ser numérico. \n';
        } else {
            document.querySelector('input[name="orderId"]').classList.remove('border-red');
        }

        if (!isValid) {
            return toast.error(message, { position: toast.POSITION.TOP_CENTER });
        }

        const resp = await dispatch(startDeleteOrderEvent(orderId));

        if (resp.ok) {
            toast.success(<ToastSuccess text={resp.msg} />, { position: toast.POSITION.TOP_CENTER });
        } else {
            toast.error(<ToastError text={resp.msg} />, { position: toast.POSITION.TOP_CENTER });
        }

        setFormValuesOrder({ orderId: '' });

    }


    return (
        <div className='admin-screen animate__animated animate__fadeIn'>
            {idUser && <AdminModal
                setIdUser={setIdUser}
                idUser={idUser}
                users={users}
                setUsers={setUsers} />}
            <ToastContainer />
            <h1>Administración</h1>

            <div className="users-main-wrapper">

                <div className="wrapper-users-table">
                    <h3>Usuarios registrados</h3>
                    <div className="users-header">
                        <div>ID</div>
                        <div>Email</div>
                        <div>Nombre</div>
                        <div>Admin</div>
                        <div>Activo</div>
                        <div>Opciones</div>
                    </div>

                    {
                        users && users.length > 0 &&

                        users.map(user => {

                            return (
                                <div key={user.id} className="users-body">
                                    <div>{user.id}</div>
                                    <div>{user.email}</div>
                                    <div>{user.name}</div>
                                    <div>{user.admin ? "Si" : "No"}</div>
                                    <div className="user-status">
                                        {user.active
                                            ? <i className="fas fa-check"></i>
                                            : <i className="fas fa-times"></i>
                                        }
                                    </div>
                                    <div className="options-user">
                                        <i className="fas fa-wrench" onClick={() => { handleUpdateUser(user.id) }}></i>

                                        {user.active
                                            ? <i className="fas fa-user-minus" onClick={() => { handleStartDeleteUser(user.id, user.name) }}></i>
                                            : <i className="fas fa-user-plus" onClick={() => { handleStartDeleteUser(user.id, user.name) }}></i>
                                        }

                                    </div>
                                </div>
                            );
                        })
                    }
                </div>

                <div className="forms-wrapper" onSubmit={handleSubmitUser}>
                    <form className="form-add-users">
                        <h3>Agregar nuevo usuario</h3>

                        <div className="main-wrapper-inputs">

                            <div className="wrapper-input">
                                <label>Admin: </label>
                                <Switch
                                    onChange={handleInputSwitch}
                                    checked={isAdmin}
                                    onColor="#ffa600"
                                    height={20}
                                    width={42} />
                            </div>

                            <div className="wrapper-input">
                                <label>Email: </label>
                                <input
                                    type="text"
                                    name="email"
                                    value={email}
                                    onChange={handleInputChangeUser}
                                    placeholder="Email"
                                    autoComplete="off"
                                />
                            </div>

                            <div className="wrapper-input">
                                <label>Nombre: </label>
                                <input
                                    type="name"
                                    name="name"
                                    value={name}
                                    onChange={handleInputChangeUser}
                                    placeholder="Nombre"
                                    autoComplete="off"
                                />
                            </div>

                            <div className="wrapper-input">
                                <label>Password: </label>
                                <input
                                    type="password"
                                    name="password"
                                    value={password}
                                    onChange={handleInputChangeUser}
                                    placeholder="Password"
                                    autoComplete="off"
                                />
                            </div>

                        </div>

                        <div className="main-wrapper-submit">
                            <button className="btn btn-order" type="submit">Crear</button>
                        </div>

                    </form>

                    <form className="form-remove-orders" onSubmit={handleStartDeleteOrder}>
                        <h3>Eliminar orden</h3>

                        <div className="wrapper-input">
                            <label>Número Orden: </label>
                            <input
                                type="text"
                                value={orderId}
                                onChange={handleInputChangeOrder}
                                placeholder="Número"
                                autoComplete="off"
                                name="orderId"
                            />
                        </div>
                        <div className="main-wrapper-submit">
                            <button className="btn btn-order-cancel" type="submit">Eliminar</button>
                        </div>
                    </form>
                </div>

            </div>

        </div>
    )
}
