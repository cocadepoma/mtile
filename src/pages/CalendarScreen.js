import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'

// Big-Calendar dependencies
import { Calendar, momentLocalizer } from "react-big-calendar";
import moment from "moment";

import "react-big-calendar/lib/css/react-big-calendar.css";
import { messages } from '../helpers/calendar-messages';

// Import Spanish texts for days and months
import "moment/locale/es";
import { CalendarEvent } from '../components/calendar/CalendarEvent';
import { CalendarModal } from '../components/calendar/CalendarModal';
import { uiOpenModal } from '../actions/ui';
import { setActiveEvent } from '../actions/calendar';
import { getEventColor } from '../helpers/getEventColor';
import { disableScroll } from '../helpers/disable-enable-scroll';
moment.locale("es");
const localizer = momentLocalizer(moment);


export const CalendarScreen = () => {

    // Get events array from Redux Store
    const { events } = useSelector(state => state.calendar);

    const dispatch = useDispatch();

    // Set active event on Redux Store calendar: events[], activeEvent = event
    // then => Open modal
    const handleEventClick = (event) => {
        dispatch(setActiveEvent(event));
        dispatch(uiOpenModal());
        disableScroll();
    };

    // Calendar events style
    const eventStyleGetter = (event, start, end, isSelected) => {

        const { backgroundColor, textColor } = getEventColor(event);

        const style = {
            backgroundColor: backgroundColor,
            borderRadius: "5px",
            opacity: 0.8,
            display: "block",
            color: textColor,
        };
        return {
            style,
        };
    };
    // Save last position in localStorage
    const [lastView, setLastView] = useState(localStorage.getItem("lastView") || "month");

    // When the view changes, save the position in localStorage
    const onViewChange = (e) => {
        setLastView(e);
        localStorage.setItem("lastView", e);
    };

    return (
        <div className='calendar-screen animate__animated animate__fadeIn'>
            <h1>Calendario</h1>
            <Calendar
                defaultDate={moment().toDate()}
                defaultView="month"
                events={events}
                localizer={localizer}
                eventPropGetter={eventStyleGetter}
                messages={messages}
                components={{ event: CalendarEvent }}
                onSelectEvent={handleEventClick}
                onView={onViewChange}
                view={lastView}
                startAccessor='start'
                endAccessor='end'
            />
            <CalendarModal />
        </div>
    )
}
