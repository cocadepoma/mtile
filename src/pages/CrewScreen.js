
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setActiveTechnician } from '../actions/technician';
import { uiOpenModal } from '../actions/ui';

import { ButtonNew } from '../components/ui/ButtonNew';
import { CrewModal } from '../components/crew/CrewModal';
import { disableScroll } from '../helpers/disable-enable-scroll';

export const CrewScreen = () => {

    const { technicians } = useSelector(state => state.crew);
    const { admin } = useSelector(state => state.auth);

    const dispatch = useDispatch();

    const openModalTechnician = (e) => {

        const id = e.target.attributes[1].value;
        const clickedTechnician = technicians.find((technician) => {
            return technician.id.toString() === id;
        });

        dispatch(setActiveTechnician(clickedTechnician));
        dispatch(uiOpenModal());
        disableScroll();
    }

    // Button Add new Technician
    const iconData = {
        icon: <i className="fas fa-user-plus"></i>,
        className: 'button-new technician'
    }

    return (
        <>
            <h1 className="technician-h1">Técnicos</h1>
            <div className="crew-wrapper animate__animated animate__fadeIn">

                {
                    technicians.map((technician) => {

                        return (
                            <div key={technician.id} className="technician-wrapper">

                                <div className="technician-name-surname">
                                    <span>{technician.surname},</span>
                                    <span>{technician.name}</span>
                                    <span></span>
                                </div>

                                <img src={technician.image === null
                                    ? `${process.env.PUBLIC_URL}/assets/images/user-default.png`
                                    : `${process.env.PUBLIC_URL}/assets/images/${technician.image}`}
                                    alt={`${technician.name}`}
                                />

                                {
                                    admin
                                    &&
                                    <div className="wrapper-technician-button">
                                        <button className="btn btn-detail-crew" data-id={technician.id} onClick={openModalTechnician}>Ver más</button>
                                    </div>
                                }
                            </div>
                        )
                    })
                }
            </div >
            {
                admin && <ButtonNew iconData={iconData} page="crew" />
            }
            <CrewModal />
        </>
    )
}
