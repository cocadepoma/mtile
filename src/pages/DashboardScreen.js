import React, { useEffect, useState } from 'react'
import { useSelector } from 'react-redux';
import { TableDashboard } from '../components/dashboard/TableDashboard'
import { getColumnsTableDashboard } from '../helpers/getColumnsTables';
import { getOrdersStillOpen } from '../helpers/getOrdersStillOpen';
import Chart from "react-apexcharts";
import { useDispatch } from 'react-redux';
import { startDeleteWarning, startLoadWarnings } from '../actions/warnings';
import { dataBar, dataDonut, optionsBar, optionsDonut } from '../helpers/mockDataCharts';
import { ModalToastify } from '../components/ui/ModalToastify';
import { toast, ToastContainer } from 'react-toastify';
import { ModalWarning } from '../components/dashboard/ModalWarning';
import { uiOpenModal } from '../actions/ui';
import { disableScroll } from '../helpers/disable-enable-scroll';
import { Link } from 'react-router-dom';


export const DashboardScreen = () => {

    const dispatch = useDispatch();
    const { warnings } = useSelector(state => state.warning);
    const [showWarningModal, setShowWarningModal] = useState(false);


    useEffect(() => {
        dispatch(startLoadWarnings());
    }, [dispatch]);

    const columns = getColumnsTableDashboard;
    const { events } = useSelector(state => state.calendar);
    const ordersStillOpen = getOrdersStillOpen(events);

    const handleAddWarning = () => {
        setShowWarningModal(true);
        disableScroll();
        dispatch(uiOpenModal());
    };

    const showDeleteWarning = (id, i) => {

        toast.warn(<ModalToastify
            handleDeleteItem={() => handleDeleteWarning(id)}
            code={warnings[i].description}
            message="Estás seguro de borrar el aviso " />,
            {
                position: toast.POSITION.TOP_CENTER,
                closeOnClick: false,
                autoClose: false,
                toastId: '1'
            });

    }
    const handleDeleteWarning = (id) => {
        dispatch(startDeleteWarning(id));
    };

    return (
        <div className='animate__animated animate__fadeIn dashboard-screen'>
            <h1>DashBoard</h1>

            <div className='main-content_dashboard'>

                <div className="charts">
                    <div className="chart-wrapper">
                        <h3 className="h3-dashboard charts-header">Intervenciones por sección</h3>
                        <Chart
                            options={optionsBar}
                            series={dataBar}
                            type="bar"
                            width="100%"
                        />
                    </div>
                    <div className="chart-wrapper">
                        <h3 className="h3-dashboard charts-header">Intervenciones Ayer</h3>

                        <Chart
                            className="chart-pie"
                            options={optionsDonut}
                            series={dataDonut}
                            type="donut"
                            width="100%"
                        />
                    </div>

                </div>

                <div className="events-alerts-wrapper">
                    <div className="lasts-events-wrapper">
                        <div className="header-warnings">
                            <h3 className="h3-dashboard">
                                <span className="prueba">ó</span>rdenes abiertas
                            </h3>
                            <Link to="/neworder"><i className="fas fa-plus-circle"></i></Link>
                        </div>

                        <div className="lasts-events">
                            {ordersStillOpen.length > 0 && <TableDashboard columns={columns} data={ordersStillOpen} />}
                        </div>
                    </div>

                    <div className="lasts-alerts-wrapper">
                        <div className="header-warnings">
                            <h3 className="h3-dashboard">
                                <span className="prueba">ú</span>ltimos avisos
                            </h3>
                            <i className="fas fa-plus-circle" onClick={handleAddWarning}></i>
                        </div>

                        <div className="alerts-wrapper">

                            {warnings &&

                                warnings.map((warning, i) => {

                                    return <div key={warning.id} className="warnings">
                                        <div className="warning">
                                            <p>{warning.description}</p>
                                            <i className="far fa-trash-alt" onClick={() => showDeleteWarning(warning.id, i)}></i>
                                        </div>
                                    </div>

                                })
                            }
                        </div>
                    </div>
                </div>

                {showWarningModal && <ModalWarning setShowWarningModal={setShowWarningModal} />}

                <ToastContainer />
            </div>
        </div>
    )
}
