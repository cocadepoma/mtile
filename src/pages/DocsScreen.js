import React, { useEffect, useState } from 'react';
import { PdfLoader } from '../components/doc/PdfLoader';
import { useDispatch, useSelector } from 'react-redux';
import { setActiveDoc, startLoadFactory } from '../actions/factory';
import { getSectionsByFactoryId } from '../helpers/getSectionsByFactoryId';
import { getDocsBySectionId } from '../helpers/getDocsBySectionId';
import ClickAwayListener from 'react-click-away-listener';
import { disableScroll } from '../helpers/disable-enable-scroll';
import { PdfModal } from '../components/doc/PdfModal';
import { getDocById } from '../helpers/getDocById';
import { uiOpenModal } from '../actions/ui';
import { ButtonNew } from '../components/ui/ButtonNew';
import { PdfModalAdd } from '../components/doc/PdfModalAdd';



export const DocsScreen = () => {

    const { factories } = useSelector(state => state.factory);
    const { sections } = useSelector(state => state.factory);
    const { docs } = useSelector(state => state.factory);
    const { admin } = useSelector(state => state.auth);
    const dispatch = useDispatch();

    // Array of sections from factory
    const [showSections, setShowSections] = useState([]);
    // Array of docs from a section
    const [showPdf, setShowPdf] = useState([]);
    // State to know if the Select is opened or closed
    const [isRotateFactory, setIsRotateFactory] = useState(false);
    // State to know if the Select is opened or closed
    const [isRotateSection, setIsRotateSection] = useState(false);
    // Breadcrumb
    const [optionsSelected, setOptionsSelected] = useState({ factory: '', section: '' });

    const [showModalDoc, setShowModalDoc] = useState(false);

    // This function depends on ClickAwayListener. Will close customs selects 
    // if the user clicks outside the component
    const handleClickAway = () => {
        isRotateFactory && toggleFactory();
        isRotateSection && toggleSection();
    };

    // Load and fecth factories, sections, machines and docs from DB
    useEffect(() => {
        dispatch(startLoadFactory());
    }, [dispatch]);

    // Reset data from custom select and breadcrumbs
    const resetData = () => {
        setOptionsSelected({ factory: '', section: '' });
        setShowSections([]);
        setShowPdf([]);
    }


    // When click Factory select, will set breadcrumb text,
    // close select, will show sections and set the array of documents to an empty array. 
    const handleClickFactory = ({ target }) => {

        // Set factory breadcrumb
        setOptionsSelected({ factory: target.innerText, section: '' });

        // Get sections from factory
        const id = target.attributes[0].value;
        const factorySections = getSectionsByFactoryId(id, sections);

        // Show sections select
        setShowSections(factorySections);
        // Hide previous documents by setting an empty array
        setShowPdf([]);
    }

    const handleClickSection = ({ target }) => {

        // Set section breadcrumb
        setOptionsSelected({ ...optionsSelected, section: target.innerText });

        // Get docs from section
        const id = target.attributes[0].value;
        const sectionDocs = getDocsBySectionId(id, docs);

        // Show PDF's
        setShowPdf(sectionDocs);
    }

    const handleDocClick = (id) => {
        const docData = getDocById(id, docs);
        disableScroll();
        setShowModalDoc(true);
        dispatch(setActiveDoc(docData));
        dispatch(uiOpenModal());
    }

    // Show & Hide Custom Select for Factories. Removing and Adding additional css classes.
    const toggleFactory = () => {
        const clicker = document.querySelector('.clicker');
        const clickerSelect = document.querySelector('.factory-selector');

        if (!clicker || !clickerSelect) {
            return;
        }

        if (!isRotateFactory) {
            clicker.classList.add('rotate');
            clickerSelect.classList.add('showlist');
        } else {
            clicker.classList.remove('rotate');
            clickerSelect.classList.remove('showlist');
        }

        setIsRotateFactory(!isRotateFactory);

    }

    // Show & Hide Custom Select for Sections. Removing and Adding additional css classes.
    const toggleSection = () => {

        if (!isRotateSection) {
            document.querySelector('.clicker-2').classList.add('rotate');
            document.querySelector('.section-selector').classList.add('showlist');
        } else {
            document.querySelector('.clicker-2').classList.remove('rotate');
            document.querySelector('.section-selector').classList.remove('showlist');
        }

        setIsRotateSection(!isRotateSection);

    }


    // Button Add new Item
    const iconData = {
        icon: <i className="fas fa-file-upload"></i>,
        className: 'button-new doc',
    }


    return (
        <div className="docs-main-container">
            <h1 className="h1-docs">Documentaciones</h1>

            {/*  breadcrumb */}
            <p className={`doc-breadcrumb ${optionsSelected.factory ? 'show-bread animate__animated animate__fadeIn animate__fast' : ''}`}>
                Factoría {optionsSelected.factory && `${optionsSelected.factory}`}
                {optionsSelected.factory && <i className="fas fa-caret-right bread-arrow"></i>}

                {optionsSelected.section && `${optionsSelected.section}`}
            </p>

            {/*  Factories Custom select */}
            <div className="doc-selectors">
                <div className="factory-header-selects animate__animated animate__fadeIn animate__fast">
                    <ClickAwayListener onClickAway={handleClickAway}>
                        <div className="factory-header" onClick={toggleFactory}>
                            <div className="grid-name">
                                <i className="fas fa-industry"></i>
                                <h4>Factorías</h4>
                            </div>
                            <i className="fas fa-sort-down clicker" onClick={toggleFactory}></i>
                        </div>
                    </ClickAwayListener>

                    <div className="factory-selector">
                        <ul onClick={handleClickFactory}>
                            {
                                factories.map
                                    (
                                        factory =>
                                            <li key={factory.id} value={factory.id}>
                                                {factory.name}
                                            </li>
                                    )
                            }
                        </ul>
                    </div>
                </div>

                {/*  Sections Custom select */}
                {
                    showSections.length > 0
                    &&
                    <div className="section-header-selects animate__animated animate__fadeIn animate__fast">
                        <ClickAwayListener onClickAway={handleClickAway}>
                            <div className="section-header" onClick={toggleSection}>
                                <div className="grid-name">
                                    <i className="fas fa-project-diagram"></i>
                                    <h4>Secciones</h4>
                                </div>
                                <i className="fas fa-sort-down clicker-2" onClick={toggleSection}></i>
                            </div>
                        </ClickAwayListener>

                        <div className="section-selector">
                            <ul onClick={handleClickSection}>

                                {
                                    showSections.length > 0
                                    &&
                                    showSections.map
                                        (
                                            section =>
                                                <li key={section.id} value={section.id}>
                                                    {section.name}
                                                </li>
                                        )
                                }
                            </ul>
                        </div>
                    </div>
                }
            </div>

            {/*  Documents */}

            {
                (showPdf.length > 0)
                    ? <div className="pdf-wrapper">
                        {showPdf.map(pdf =>
                            <div key={pdf.id} className="single-pdf-wrapper animate__animated animate__fadeIn" onClick={() => { handleDocClick(pdf.id) }}>
                                <PdfLoader pdf={pdf} />
                            </div>
                        )}
                    </div>

                    : (optionsSelected.section && showPdf.length === 0) && <h3 className="h3-docs animate__animated animate__zoomIn">No se ha encontrado documentación</h3>
            }

            {showModalDoc && <PdfModal setShowModalDoc={setShowModalDoc} resetData={resetData} />}

            {!showModalDoc && <PdfModalAdd resetData={resetData} />}


            {
                admin && <ButtonNew iconData={iconData} />
            }

        </div>
    );

}
