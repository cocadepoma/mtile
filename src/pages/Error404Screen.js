import React from 'react'
import { Link, useHistory } from 'react-router-dom'

export const Error404Screen = () => {

    const history = useHistory();

    console.log(history)
    return (
        <div>
            error 404ddddddsss
            <Link to='/'>Volver al Dashboard</Link>
            <button onClick={() => { history.goBack() }}>Volver atr√°s</button>
        </div>
    )
}
