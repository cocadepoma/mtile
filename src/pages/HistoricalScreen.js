import React, { useEffect, useState } from 'react'
import { getColumnsTableHistorical } from '../helpers/getColumnsTables';
import { useSelector } from 'react-redux';
import { TableHistorical } from '../components/historical/TableHistorical';



export const HistoricalScreen = () => {

    const columns = getColumnsTableHistorical;
    const { events } = useSelector(state => state.calendar);
    const [orderedEvents, setOrderedEvents] = useState([]);

    useEffect(() => {
        const aux = events.sort((a, b) => b.start - a.start);
        setOrderedEvents(aux);
    }, [events]);

    return (
        <div className='historical-main-container animate__animated animate__fadeIn'>
            <h1 className='h1-historical'>HistÃ³rico</h1>
            {
                orderedEvents.length > 0 && <TableHistorical columns={columns} data={events} />
            }
        </div>
    )
}
