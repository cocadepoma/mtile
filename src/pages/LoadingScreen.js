import React from 'react'
import { Loading } from '../components/ui/Loading'

export const LoadingScreen = () => {
    return (
        <div className="loading-screen-wrapper animate__animated animate__fadeIn animate__faster">
            <Loading />
        </div>
    )
}
