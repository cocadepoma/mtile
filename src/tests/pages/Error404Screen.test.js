import React from 'react';
import { Error404Screen } from '../../pages/Error404Screen';
import { render, fireEvent, cleanup, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
// import { prettyDOM } from '@testing-library/dom';
import { Router, BrowserRouter as RouterP } from 'react-router-dom';
import { createMemoryHistory, createBrowserHistory } from 'history';

describe('tests on Error404Screen', () => {

    afterEach(() => {
        cleanup();
    });

    test('should render correctly', () => {
        const history = createBrowserHistory();

        const { container, getByText } = render(
            <Router history={history}>
                <Error404Screen />
            </Router>
        );

        expect(container).toMatchSnapshot();
        getByText('Volver al Dashboard');
        getByText('Volver atrÃ¡s');

    });

    test('should redirect to dashboard if user clicks the button', () => {
        const history = createBrowserHistory();
        const route = '/dashboard/eue';
        history.push(route);

        const { getByText } = render(
            <Router history={history}>
                <Error404Screen />
            </Router>
        );
        expect(global.window.location.pathname).toEqual('/dashboard/eue');

        const button = getByText('Volver al Dashboard');
        fireEvent.click(button);
        expect(global.window.location.pathname).toEqual('/');
    });


});
