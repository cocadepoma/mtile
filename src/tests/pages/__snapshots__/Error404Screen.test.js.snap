// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on Error404Screen should render correctly 1`] = `
<div>
  <div
    class="screen-404 animate__animated animate__fadeIn"
  >
    <div
      class="bg-black"
    >
      <img
        alt="bg"
        class="bg-404"
        src="/assets/images/404.jpg"
      />
      <div
        class="info-404"
      >
        <h1>
          404
        </h1>
        <p
          class="text-info"
        >
          Página no encontrada
        </p>
        <div
          class="buttons-404"
        >
          <a
            href="/"
          >
            Volver al Dashboard
          </a>
          <p>
            Volver atrás
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
